CedarJava/src/test/java/com/cedarpolicy/EntityValidationTests.java

```diff
@@ -0,0 +1,118 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.BeforeAll;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+
+import com.cedarpolicy.model.EntityValidationRequest;
+import com.cedarpolicy.model.EntityValidationResponse;
+
+import com.cedarpolicy.model.exception.AuthException;
+import com.cedarpolicy.model.schema.Schema;
+import com.cedarpolicy.model.slice.Entity;
+import com.cedarpolicy.pbt.EntityGen;
+import com.cedarpolicy.value.EntityTypeName;
+import com.cedarpolicy.value.PrimBool;
+
+
+/**
+ * Tests for entity validator
+ */
+public class EntityValidationTests {
+    private static EntityGen entityGen;
+    private static AuthorizationEngine engine;
+
+    /**
+     * Test that a valid entity is accepted.
+     */
+    @Test
+    public void testValidEntity() throws AuthException {
+        Entity entity = EntityValidationTests.entityGen.arbitraryEntity();
+
+        EntityValidationRequest r = new EntityValidationRequest(
+                ROLE_SCHEMA, List.of(entity));
+
+        EntityValidationResponse result = engine.validateEntities(r);
+
+        assertEquals(result.getMessage(), "Entities parsed correctly");
+    }
+
+    /**
+     * Test that an entity with an attribute not specified in the schema throws an exception.
+     */
+    @Test
+    public void testEntityWithUnknownAttribute() throws AuthException {
+        Entity entity = EntityValidationTests.entityGen.arbitraryEntity();
+        entity.attrs.put("test", new PrimBool(true));
+
+        EntityValidationRequest request = new EntityValidationRequest(
+                ROLE_SCHEMA, List.of(entity));
+
+
+        AuthException exception = assertThrows(AuthException.class, () -> engine.validateEntities(request));
+        assertTrue(
+                exception.message.orElse("")
+                        .startsWith("Bad request: error during entity deserialization: attribute `test` on "),
+                "Expected exception message to start with 'Bad request: error during entity deserialization: attribute `test` on'");
+    }
+
+    /**
+     * Test that entities with a cyclic parent relationship throw an exception.
+     */
+    @Test
+    public void testEntitiesWithCyclicParentRelationship() throws AuthException {
+        // Arrange
+        Entity childEntity = EntityValidationTests.entityGen.arbitraryEntity();
+        Entity parentEntity = EntityValidationTests.entityGen.arbitraryEntity();
+
+        // Create a cyclic parent relationship between the entities
+        childEntity.parentsEUIDs.add(parentEntity.getEUID());
+        parentEntity.parentsEUIDs.add(childEntity.getEUID());
+
+        EntityValidationRequest request = new EntityValidationRequest(
+                ROLE_SCHEMA,
+                List.of(parentEntity, childEntity));
+
+        // Act & Assert
+        AuthException exception = assertThrows(AuthException.class, () -> engine.validateEntities(request));
```

Comment:
Same question as above

Reply:
`InternalException` seems like the best fit.

Reply:
Same comment as above